font/src/ft/mod.rs

```diff
@@ -278,12 +316,19 @@ impl FreeTypeRasterizer {
                 }
             }
 
+            // Reuse the original fontkey if you're reloading the font
+            let key = if let Some(key) = key {
+                key
+            } else {
+                FontKey::next()
+            };
```

Comment:
Imagine you're a downstream crate, and you request 3 fonts, `A`, `B`, where `A` == `B`. So I'm loading font `A` and getting FontKey `1`, then I'm trying to load font `B` which is the same as `A`, so if I don't reuse FontKey, you'll invalidate a FontKey for `A` and generate a new one for `B`, so next time your downstream will query the font by `FontKey` it wont get the right one (I was fixing this issue during implementation). So you don't want your internal caches affect your public API. Technically this should also applies to fallback fonts, but since alacritty is the only one downstream right now, and wiping the cache I thought that it's fine for now.